external/storm-jdbc/src/main/java/org/apache/storm/jdbc/bolt/JdbcBolt.java

```diff
@@ -0,0 +1,81 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.jdbc.bolt;
+import backtype.storm.topology.OutputFieldsDeclarer;
+import backtype.storm.tuple.Tuple;
+import org.apache.storm.jdbc.common.Column;
+import org.apache.storm.jdbc.mapper.JdbcMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Basic bolt for writing to any Database table.
+ * <p/>
+ * Note: Each JdbcBolt defined in a topology is tied to a specific table.
+ */
+public class JdbcBolt extends AbstractJdbcBolt {
+    private static final Logger LOG = LoggerFactory.getLogger(JdbcBolt.class);
+
+    private String tableName;
+    private JdbcMapper jdbcMapper;
+
+    public JdbcBolt withConfigKey(String configKey) {
```

Comment:
OK So I see two ways to do the Hikari configuration.
The current way.
```
Map<String, String> jdbcConf = new HashMap<String, String>();
jdbcConf.put("a","b");
stormConf.put("something", jdbcConf);
bolt.withConfigKey("something");
```

or we could just pass the jdbc config to the Bolt, and not send it through the topology config first.

```
Map<String, String> jdbcConf = new HashMap<String, String>();
jdbcConf.put("a","b");
bolt.withConfig(jdbcConf);
``` 
I don't understand why we need the extra step of storing a config inside another config, and providing a layer of indirection.

As for constructor values vs builder values, it is a question of required vs optional parameters.

```
Map<String, String> jdbcConf = new HashMap<String, String>();
jdbcConf.put("a","b");
stormConf.put("something", jdbcConf);
//oops forgot this bolt.withConfigKey("something");
```
results in an NPE after the topology is launched.  If it were a constructor parameter.
```
Map<String, String> jdbcConf = new HashMap<String, String>();
jdbcConf.put("a","b");
stormConf.put("something", jdbcConf);
new Bolt(/*forgot "something"*/);
```
is a compile time error.  I am fine with having a builder pattern for anything that is optional, or even a way to override/extend require values.  I see it as defensive programming.